pkg/kubernetes/kubernetes.go

```diff
@@ -234,15 +234,12 @@ func (h *Client) LabelNodeAsMaster(ctx context.Context, name string, taintNoSche
 		return fmt.Errorf("failed to marshal unmodified node %q into JSON: %w", n.Name, err)
 	}
 
-	n.Labels[constants.LabelNodeRoleMaster] = ""
 	n.Labels[constants.LabelNodeRoleControlPlane] = ""
 
 	taintIndex := -1
 
-	// TODO: with K8s 1.21, add new taint LabelNodeRoleControlPlane
-
 	for i, taint := range n.Spec.Taints {
-		if taint.Key == constants.LabelNodeRoleMaster {
```

Comment:
I guess we should match on both here - in case of an upgrade, the taint will be `master`, and we want to convert it to taint `control-plane`